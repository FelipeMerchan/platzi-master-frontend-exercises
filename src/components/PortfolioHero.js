import React from 'react'

import './styles/PortfolioHero.css'
import title from '../images/portfolio-felipe-merchan.png'

function PortfolioHero() {
  return (
    <section className="PortfolioHero">
      <figure className="PortfolioHero__imageContainer">
        <img className="PortfolioHero__image" width="350" src={ title } />
      </figure>
      <h1 className="PortfolioHero__title">Mira mis últimos diseños y proyectos. Oh, por cierto, <span>¡Bienvenido!</span></h1>
    </section>
  )
}

export default PortfolioHero