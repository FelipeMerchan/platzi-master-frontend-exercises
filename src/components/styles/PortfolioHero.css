.PortfolioHero {
  max-width: 600px;
  margin-block-start: var(--space);
  margin-block-end: var(--space-2);
  font-family: var(--font-4);
}

.PortfolioHero__title {
  margin-block-start: var(--space);
  font-size: 30px;
  color: #c5345c;
  line-height: 1;
  font-weight: var(--bold);
}

.PortfolioHero__title>span {
  color: #07080d;
}