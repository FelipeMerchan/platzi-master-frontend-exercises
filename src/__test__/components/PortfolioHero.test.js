import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import PortfolioHero from '../../components/PortfolioHero'
import title from '../../images/portfolio-felipe-merchan.png'

describe('Probar el componente <PortfolioHero />', () => {
  const portfolioHero = shallow(<PortfolioHero />)
  test('Render del componente PortfolioHero',() => {
    expect(portfolioHero.length).toEqual(1)
  })
  test('Render del tÃ­tulo con el nombre del autor', () => {
    expect(portfolioHero.find('img').prop('src')).toEqual(title)
  })
  test('Render con el className PortfolioHero__title', () => {
    expect(portfolioHero.find('h1').hasClass('PortfolioHero__title')).toBe(true)
  })
})

describe('PortfolioHero Snapshot', () => {
  const portfolioHero = create(<PortfolioHero />)
  test('Comprobar la UI del componente PortfolioHero', () => {
    expect(portfolioHero.toJSON()).toMatchSnapshot()
  })
})