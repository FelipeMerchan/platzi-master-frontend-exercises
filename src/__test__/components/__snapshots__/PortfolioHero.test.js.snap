// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioHero Snapshot Comprobar la UI del componente PortfolioHero 1`] = `
<section
  className="PortfolioHero"
>
  <figure
    className="PortfolioHero__imageContainer"
  >
    <img
      alt="Portafolio de Felipe Merchan"
      className="PortfolioHero__image"
      src="portfolio-felipe-merchan.png"
      width="350"
    />
  </figure>
  <h1
    className="PortfolioHero__title"
  >
    Mira mis últimos diseños y proyectos. Oh, por cierto, 
    <span>
      ¡Bienvenido!
    </span>
  </h1>
</section>
`;
